src/Operation/BulkWrite.php

```diff
@@ -282,17 +297,25 @@ public function __construct($databaseName, $collectionName, array $operations, a
      * @see Executable::execute()
      * @param Server $server
      * @return BulkWriteResult
-     * @throws UnsupportedException if collation is used and unsupported
+     * @throws UnsupportedException if array filters or collation is used and unsupported
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
     public function execute(Server $server)
     {
+        if ($this->isArrayFiltersUsed && ! \MongoDB\server_supports_feature($server, self::$wireVersionForArrayFilters)) {
+            throw UnsupportedException::arrayFiltersNotSupported();
+        }
+
         if ($this->isCollationUsed && ! \MongoDB\server_supports_feature($server, self::$wireVersionForCollation)) {
             throw UnsupportedException::collationNotSupported();
         }
 
         $options = ['ordered' => $this->options['ordered']];
 
+        if (isset($this->options['arrayFilters'])) {
```

Comment:
This looks like dead code, as "arrayFilters" isn't a BulkWrite option but rather appears within args for individual statements in the `$operations` constructor argument.